services/cache2/provider.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package cache2
+
+import "time"
+
+// CacheOptions contains options for initializaing a cache
+type CacheOptions struct {
+	Size                   int
+	DefaultExpiry          time.Duration
+	Name                   string
+	InvalidateClusterEvent string
+}
+
+// Provider is a provider for Cache
+type Provider interface {
+	// NewCache creates a new cache with given options.
+	NewCache(opts *CacheOptions) Cache
+	// Connect opens a new connection to the cache using specific provider parameters.
+	Connect()
```

Comment:
Let's also add `error` returns for both Connect and Close whenever we add external cache support later.

Reply:
make sense. will change that.